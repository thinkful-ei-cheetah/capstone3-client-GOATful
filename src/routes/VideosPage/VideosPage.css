.my-videos-container {
  margin: auto; 
  padding: 0 1em;
  margin-top:0px;
  max-width: 1000px;
  display: grid;
  grid-gap: .5em;
  grid-template-columns: repeat(auto-fill, minmax(300px, 1fr));

}

.videos-page > .btn-container {
  margin-top: 1rem;
  display: flex;
  justify-content: space-around;
  margin-bottom: 0px;
  padding-bottom: 0px;
}

.previous, .next {
  width: 85px;
}

.add-video-header {
  text-align: center;
}

.no-videos-text{
  width: 300px;
  position: absolute;
  left: 0;
  right: 0;
  margin: auto;
  top:40%
}