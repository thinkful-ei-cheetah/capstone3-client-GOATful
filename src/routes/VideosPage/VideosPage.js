import React, { Component } from 'react';
import VideoItem from '../../components/VideoItem/VideoItem';
import './VideosPage.css';
import Loader from '../../components/Loader/Loader'
import VideoModalForm from '../../components/VideoModalForm/VideoModalForm'
import VideoService from '../../services/video-api';
import FAB from '../../components/FAB/FAB'
import { withAppContext } from '../../contexts/AppContext'
import InfiniteScroll from 'react-infinite-scroll-component';

class Videos extends Component {

  state = {
    videos: [],
    isLoading: true,
    modalIsOpen: false,
    //for infinite scroll
      currentPage: 1,
      lastPage: null,
      hasMoreVideosForScroll: true
  }

  async componentDidMount() {
    this.getVideoList()
  } 

  deleteVideo = async videoId => {
    try{
      this.setState({isLoading: true })
      await VideoService.deleteVideo(videoId)
      this.getVideoList(true)

    } catch(err){ 
      this.setState({ isLoading: false }, this.props.appContext.setAppError(err.message))
    }
  }

  openModal = () => {
    this.setState({modalIsOpen: true})  
  }

  closeModal = () => {
    this.setState({modalIsOpen: false})
  }

  renderVideos() {
    const { videos } = this.state;
    return videos.map(video => 
      <VideoItem
        deleteVideo={this.deleteVideo}
        video={video}
        key={video.id}
        getVideoList={this.getVideoList}
      />
    )
  }

  getVideoList = async (resetScroll = false) => {

    if (resetScroll){
      this.setState({
        isLoading: true,
        currentPage : 1,
        lastPage: null,
        hasMoreVideosForScroll: true,
        videos: [] 
      })
    } else {
      this.setState({
        isLoading: true
      })
    }

    try {
      const videos = await VideoService.getVideos(this.state.currentPage++);
      this.setState({ videos: [...this.state.videos, ...videos.data],
        isLoading: false, 
        lastPage: videos.last_page
       });

       if (this.state.currentPage === this.state.lastPage + 1 ){
        this.setState({
          hasMoreVideosForScroll: false,
        })
      }
    } catch(err) {
      this.setState({ isLoading: false }, this.props.appContext.setAppError(err.message))
    }
  }

  displayVideos = () => {
    if (this.state.videos.length){
      return (
        <div>
          <InfiniteScroll
            dataLength={this.state.videos.length} //This is important field to render the next data
            next={this.getVideoList}
            hasMore={this.state.hasMoreVideosForScroll}
            loader={<h4>Grabbing Videos...</h4>}
            endMessage={
              <p style={{textAlign: 'center'}}>
                <b>No more videos</b>
              </p>
            }
          >
            <div className='my-videos-container'>
              <h2 id='page-title'>Your Video Dashboard</h2>
              {this.renderVideos()}
            </div>
          </InfiniteScroll>
        </div>
      )
    } else return <p className="no-videos-text">You currently have 0 videos</p>
  }

  render() {
    return (
      <section className='videos-page page'>
        <Loader isLoading={this.state.isLoading} />
        <FAB onClick={this.openModal}/>
        <VideoModalForm
          history={this.props.history}
          isOpen={this.state.modalIsOpen}
          onRequestClose={this.closeModal}
          action='new'
        />
        {this.displayVideos()}
      </section>
    );
  }
}

export default withAppContext(Videos)