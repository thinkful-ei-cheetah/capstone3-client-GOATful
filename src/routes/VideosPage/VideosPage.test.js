import React from 'react';
import ReactDOM from 'react-dom';
import VideosPage from './VideosPage';
import renderer from 'react-test-renderer';


describe('VideosPage Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(< VideosPage />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<VideosPage/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
})
