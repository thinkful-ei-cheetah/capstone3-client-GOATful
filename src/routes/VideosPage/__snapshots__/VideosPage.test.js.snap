// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideosPage Component renders the UI as expected 1`] = `
<section
  className="videos-page page"
>
  <button
    className="fab"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-plus fa-w-14 fab-icon"
      data-icon="plus"
      data-prefix="fas"
      focusable="false"
      onClick={[Function]}
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </button>
  <p
    className="no-videos-text"
  >
    You currently have 0 videos
  </p>
</section>
`;
